src/Templates/src/templates/maui-blazor/.template.config/template.json

```diff
@@ -85,6 +85,24 @@
           "toLower": true
         }
       },
+      "nameToAppId": {
+        "type": "generated",
+        "generator": "regex",
+        "dataType": "string",
+        "parameters": {
+          "source": "nameToLower",
+          "steps": [
+            {
+              "regex": "\\s",
+              "replacement": "_"
+            },
+            {
+              "regex": "[^a-z0-9_]",
+              "replacement": ""
```

Comment:
I did a test with a console app `"123 Cats.Fishy+)"` and it replaced it all with `_`, so we can probably do that too. Also, a `.` is a valid app ID: `com.company.takealot.store.public` 

I also saw the winforms people did this: 

```xml
<RootNamespace Condition="'$(name)' != '$(name{-VALUE-FORMS-}safe_namespace)'">Company.ClassLibrary1</RootNamespace>
```

https://github.com/dotnet/templating/blob/main/docs/Naming-and-default-value-forms.md

I don't think we should do that as it may let other things in. Maybe you just need a single regex:

```
  "regex": "[^a-z0-9_\.]",
  "replacement": "_"
```

Reply:
Are back slashes allowed?

Reply:
Ah whoops just realized its because of dots being a special regex character